crates/services/p2p/src/service.rs

```diff
@@ -312,19 +313,14 @@ pub struct SharedState {
 }
 
 impl SharedState {
-    pub fn notify_gossip_transaction_validity<'a, T>(
+    pub fn notify_gossip_transaction_validity(
         &self,
-        message: &'a T,
+        message: GossipsubMessageInfo,
         acceptance: GossipsubMessageAcceptance,
-    ) -> anyhow::Result<()>
-    where
-        GossipsubMessageInfo: From<&'a T>,
-    {
-        let msg_info = message.into();
-
+    ) -> anyhow::Result<()> {
         self.request_sender
             .try_send(TaskRequest::RespondWithGossipsubMessageReport((
-                msg_info, acceptance,
+                message, acceptance,
```

Comment:
I'd prefer this is called `message_info` rather than `message` since it's only info/metadata of the message not the message itself